release_scripts/signassert.py

```diff
@@ -0,0 +1,63 @@
+################################################################################
+#                                                                              #
+# Copyright (C) 2011-2015, Armory Technologies, Inc.                           #
+# Distributed under the GNU Affero General Public License (AGPL v3)            #
+# See LICENSE or http://www.gnu.org/licenses/agpl.html                         #
+#                                                                              #
+################################################################################
+
+import getpass
+import os
+import sys
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
+
+from armoryengine.ALL import *
```

Comment:
I just looked at it briefly. While we could move the parser.add_argument calls in ArmoryUtils into a method called add_arguments and then call ArmoryUtils.add_arguments from ArmoryQt and armoryd, we still have a problem. We would be creating the parser itself in ArmoryQt and armoryd and therefore we would be calling parser.parse_args() in ArmoryQt and armoryd instead of ArmoryUtils, which is where we actually need CLI_OPTIONS and CLI_ARGS to make use of them.

So it seems like using the strategy at that link would involve moving everything that uses CLI_OPTIONS or CLI_ARGS into ArmoryQt and armoryd, which would cause a lot of doubling of code. Unless I am missing something.

I did try doing it (the first idea, not the moving everything that uses CLI_\*) in ArmoryQt and then tried to import CLI_OPTIONS and CLI_ARGS from ArmoryQt into ArmoryUtils. That appeared to cause a circular import issue, because ArmoryQt imports everything from ArmoryUtils, which imports CLI_* from ArmoryQt, which first requires everything be imported from ArmoryUtils and so on. Not sure how to resolve it. Even if we explicitly listed what it is that ArmoryQt requires from ArmoryUtils, there is so much in ArmoryUtils that depends on having CLI_* that I don't think it would work.

Besides all that, the flag would have to be -u, because when we were making gitian-builder capable of using any generic signing program instead of a specific script for signing with Armory, we went with the same flags that gpg uses.

I just tried adding the -u flag to ArmoryUtils and the script seems to work now. The interesting thing is that it *now* seems to pass through to the script's argparse. Because if I comment out the -u flag in signassert, but leave it in ArmoryUtils, it doesn't recognize -u. But the help text is still that of ArmoryUtils. So it seems to require the flag in ArmoryUtils just to pass the flag on to signassert.